config/application.rb

```diff
@@ -107,114 +108,31 @@ def config.database_configuration
     # issue is image_optim crashes on missing dependencies
     config.assets.image_optim = false
 
-    config.autoloader = :zeitwerk
+    require 'discourse_inflector'
+    Rails.autoloaders.each do |autoloader|
+      autoloader.inflector = DiscourseInflector.new
+      autoloader.inflector.inflect(
+        'onceoff' => 'Jobs',
+        'regular' => 'Jobs',
+        'scheduled' => 'Jobs'
+      )
+    end
 
     # Custom directories with classes and modules you want to be autoloadable.
-    config.autoload_paths += Dir["#{config.root}/app"]
-    config.autoload_paths += Dir["#{config.root}/app/jobs"]
-    config.autoload_paths += Dir["#{config.root}/app/serializers"]
-    config.autoload_paths += Dir["#{config.root}/lib"]
-    config.autoload_paths += Dir["#{config.root}/lib/common_passwords"]
-    config.autoload_paths += Dir["#{config.root}/lib/highlight_js"]
-    config.autoload_paths += Dir["#{config.root}/lib/i18n"]
-    config.autoload_paths += Dir["#{config.root}/lib/validators/"]
-
-    Rails.autoloaders.main.ignore(Dir["#{config.root}/app/models/reports"])
-    Rails.autoloaders.main.ignore(Dir["#{config.root}/lib/freedom_patches"])
-
-    def watchable_args
-      files, dirs = super
-
-      # Skip the assets directory. It doesn't contain any .rb files, so watching it
-      # is just slowing things down and raising warnings about node_modules symlinks
-      app_file_extensions = dirs.delete("#{config.root}/app")
-      Dir["#{config.root}/app/*"].reject { |path| path.end_with? "/assets" }.each do |path|
-        dirs[path] = app_file_extensions
-      end
+    config.autoload_paths << "#{root}/lib"
 
-      [files, dirs]
-    end
+    Rails.autoloaders.main.ignore("lib/tasks",
+                                  "lib/generators")
+
+    config.eager_load_paths << "#{root}/lib"
```

Comment:
Just learnt about this, there are some files in the `lib` directory which we do not need in production. Should we be eager loading them?

Reply:
In theory paths that are marked as ignored for the autoloaders are also ignored for eager loaded paths. It’s indeed the case for `lib/generators` but for tasks it seems some code is eager loading them at some point.